testing/opae-c/test_init_c.cpp

```diff
@@ -60,3 +60,66 @@ TEST(init, opae_init_rel) {
   opae_release();
 }
 
+/**
+ * @test       log_message
+ *
+ * @brief      When the log level is set to message, then all errors
+ *             and messages are logged.
+ */
+TEST(init, log_message) {
+  ASSERT_EQ(0, putenv((char*)"LIBOPAE_LOG=1"));
+  opae_init();
+  testing::internal::CaptureStdout();
+  testing::internal::CaptureStderr();
+
+  OPAE_ERR("Error log.");
+  OPAE_MSG("Message log.");
+  OPAE_DBG("Debug log.");
+
+  std::string log_stdout = testing::internal::GetCapturedStdout();
+  std::string log_stderr = testing::internal::GetCapturedStderr();
+
+  EXPECT_TRUE(log_stderr.find("Error log.") != std::string::npos);
+  EXPECT_TRUE(log_stdout.find("Message log.") != std::string::npos);
+  EXPECT_FALSE(log_stdout.find("Debug log.") != std::string::npos);
+}
+
+/**
+ * @test       log_error
+ *
+ * @brief      When the log level is set to error, then only errors
+ *             are logged.
+ */
+TEST(init, log_error) {
+  ASSERT_EQ(0, putenv((char*)"LIBOPAE_LOG=0"));
+  opae_init();
+  testing::internal::CaptureStdout();
+  testing::internal::CaptureStderr();
+
+  OPAE_ERR("Error log.");
+  OPAE_MSG("Message log.");
+  OPAE_DBG("Debug log.");
+
+  std::string log_stdout = testing::internal::GetCapturedStdout();
+  std::string log_stderr = testing::internal::GetCapturedStderr();
+
+  EXPECT_TRUE(log_stderr.find("Error log.") != std::string::npos);
+  EXPECT_FALSE(log_stdout.find("Message log.") != std::string::npos);
+  EXPECT_FALSE(log_stdout.find("Debug log.") != std::string::npos);
+}
+
+/**
+ * @test       log_file
+ *
+ * @brief      When LIBOPAE_LOGFILE is specified, then the logger
+ *             will log to the specified file.
+ */
+TEST(init, log_file) {
+  ASSERT_EQ(0, putenv((char*)"LIBOPAE_LOGFILE=opae_log.log"));
+  opae_init();
+
+  struct stat buf;
+
+  EXPECT_EQ(0, stat("opae_log.log", &buf));
```

Comment:
You probably want to make sure this file doesn't exist before the test runs.

Reply:
Done. But I don't think we can remove it from inside the test once we're done.